kimu/core/intersection_tool_lines.py

```diff
@@ -0,0 +1,72 @@
+from qgis.PyQt.QtCore import QVariant
+from qgis.core import (
+    QgsFeature,
+    QgsField,
+    QgsPointXY,
+    QgsProject,
+    QgsGeometry,
+    QgsVectorLayer,
+    QgsWkbTypes,
+)
+from qgis.utils import iface
+
+from ..qgis_plugin_tools.tools.custom_logging import setup_logger
+from ..qgis_plugin_tools.tools.i18n import tr
+from ..qgis_plugin_tools.tools.resources import plugin_name
+
+LOGGER = setup_logger(plugin_name())
+
+
+class IntersectionLines:
+
+    @staticmethod
+    def __check_valid_layer(layer: QgsVectorLayer) -> bool:
+        """Checks if layer is valid"""
+        if (
+            isinstance(layer, QgsVectorLayer)
+            and layer.isSpatial()
+            and layer.geometryType() == QgsWkbTypes.LineGeometry
+        ):
+            return True
+        return False
+
+    def run(self) -> None:
+        layer = iface.activeLayer()
+        if not self.__check_valid_layer(layer):
+            LOGGER.warning(tr("Please select a line layer"), extra={"details": ""})
+            return
+
+        if len(layer.selectedFeatures()) != 2:
+            LOGGER.warning(tr("Please select two line features"), extra={"details": ""})
+            return
+
+        vl = QgsVectorLayer("Point", "temp", "memory")
+        crs = layer.crs()
+        vl.setCrs(crs)
+        pr = vl.dataProvider()
+        pr.addAttributes([QgsField("xkoord", QVariant.Double), QgsField("ykoord", QVariant.Double)])
+        vl.updateFields()
+
+        l = list()
+
+        features = layer.selectedFeatures()
+        for feat in features:
+            viiva = feat.geometry().asPolyline()
+            pxy = QgsPointXY(viiva[0])
+            pxy2 = QgsPointXY(viiva[-1])
+            l.extend([pxy.x(), pxy.y(), pxy2.x(), pxy2.y()])
+
+        # x1=l[0], y1=l[1], x2=l[2], y2=l[3], x3=l[4], y3=l[5], x4=l[6], y4=l[7]
+        x = (l[0]*((l[3]-l[1])/(l[2]-l[0]))-l[4]*((l[7]-l[5])/(l[6]-l[4]))+l[5]-l[1]) / (((l[3]-l[1])/(l[2]-l[0]))-((l[7]-l[5])/(l[6]-l[4])))
+        y = ((l[3]-l[1])/(l[2]-l[0]))*(x-l[0])+l[1]
```

Comment:
I don't doubt this works, but what happens here, i.e. what are you doing and how/which method? Should probably be explained in a comment, the current one is not that descriptive.

Reply:
This is based on analytical geometry. I determine the functions of the straight lines each of the selected line features represent (each line can be seen as a limited representation of a function determining a line which has no start and end points). After that I search for intersection point of these two functions. The equation you see here is based on

![image](https://user-images.githubusercontent.com/55135974/153037648-eec6919c-dfd9-4bc4-b25e-0bb45bccc154.png)

After determining the functions for the straight lines, the intersection point is found by setting these two functions equal and solving x. I just have solved x algebrally on paper so it is enough here to insert the values of the known 4 points and calculate the value for x.

The y value can be computed from x value and it is based on the very same two point form formula. I just have modified the basic form so that only y can be found of the left side of the equation.

Not sure if this clarified anything..

Reply:
Sounds good! :+1: It's difficult to interpret this just from the code, so I would definitely add some comment such as `convert lines to two point form and solve x`, and include a link to the webpage.